### PR TITLE
Run unit tests on Apple Silicon and bump Node version to LTS (currently 20)

### PR DESCRIPTION
**Pull request by hoechenberger**
Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by hoechenberger** [comment]
blocked by #1233

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -111,8 +111,8 @@ jobs:
 
   test:
     # For GitHub "required" CI checks, add in branch protection:
-    # 6 checks:
-    # for each OS (ubuntu, macos, windows):
+    # 7 checks:
+    # for each OS (ubuntu, macos, macos-14 windows):
```
Comment (hoechenberger) [inline .github/workflows/unit_tests.yml:115]:
```suggestion
    # for each OS (ubuntu, macos, macos-14, windows):
```

**Review by sappelhoff — APPROVED** [review]
lgtm, thanks

**Comment by hoechenberger** [comment]
@sappelhoff I will take care of the failing CI job

**Comment by hoechenberger** [comment]
@sappelhoff I think you merged a failing branch :) I'll open a follow-up PR to fix this

**Comment by sappelhoff** [comment]
urgs, sorry.

**Comment by hoechenberger** [comment]
> urgs, sorry.

No worries, happens … I think it got auto-merged because the new CI job had naturally not been added to the branch protection rules yet!

**Pull request merged**